### ISSUE TITLE
Possible to see Jasmine Version being used?

### ISSUE THREAD

**Issue by Barryrowe** [wallabyjs/public#905]
### Issue description or question
Is there a way to easily see the version of Jasmine that wallaby is using while running tests? I've got an issue I am trying to track down where Wallaby runs all my tests and they pass just fine, but Karma chokes and ends up with a bunch of cryptic errors that I can't make sense of. I believe it might be a difference in the Jasmine version being used. The best I've got is a wonderful:  "Expected spy but got Function" error, with no lines or reference source file.

I've got jasmine-core@^2.5.2 installed via NPM, which is what Karma is picking up. According to the  wallaby docs using ```testFramework: "jasmine"``` will use the latest available version. Just to test I tried using ```"jasmine@2.5.2"``` but got an error saying that version isn't supported.

### Wallaby.js configuration file

based on the Ionic2 Boilerplate you've got, updated for my project
```javascript
var wallabyWebpack = require('wallaby-webpack');

var webpackPostprocessor = wallabyWebpack({
  entryPatterns: [
    //'spec-bundle.js',
    'config/wallaby-shim.js',    
    'src/**/*spec.js'
  ]
});

module.exports = function (w) {

  return {
    files: [
      {pattern: 'src/**/*.ts', load: false},      
      {pattern: "mocks/**/*.ts", load:false},
      {pattern: 'config/wallaby-shim.ts', load: false},                  
      {pattern: 'src/**/*spec.ts', ignore: true},
      
    ],
    tests: [
      { pattern: 'src/**/*spec.ts', load: false }
    ],
    testFramework: "jasmine",
    postprocessor: webpackPostprocessor,

    bootstrap: function () {
      window.__moduleBundler.loadTests();
    }
  };
};
```

### Code editor or IDE name and version
Visual Studio Code Insiders v1.8.0-insider  (2abfc2078)

### Wallaby Plugin version 
1.0.35

### OS name and version
Windows

**Comment by ArtemGovorov** [comment]
> According to the wallaby docs using testFramework: "jasmine" will use the latest available version

It's actually "the latest supported version". For browser we ship the versions of jasmine that we support and making the most stable version as default (which is *usually* the latest shipped, but not always because sometimes we hold off marking the latest shipped version to be the default version). Versions that are shipped with wallaby core, but not marked as default, can be selected by specifying the version in the `testFramework` setting.

> just to test I tried using "jasmine@2.5.2" but got an error saying that version isn't supported

We had `jasmine@2.5.1` supported, I have just added `jasmine@2.5.2` support too in the core v1.0.334 (you'll need to specify the version as it's not marked as the default yet).

> Is there a way to easily see the version of Jasmine that wallaby is using while running tests?

Sure, in your `bootstrap` function you may just add:

```javascript
bootstrap: function () {
      console.log('Jasmine version: ' + jasmine.version); // <--
      window.__moduleBundler.loadTests();
    }
```
restart wallaby, and look into the Wallaby Console output channel for the result.

> "Expected spy but got Function" error, with no lines or reference source file.

I have searched jasmine source code and it looks like the error is thrown if a `toHaveBeenCalled` matcher is used on a function that for some reason is not being spied on. I understand that it may not provide you enough info to see why the tests are failing in karma, but at least could give a direction for an investigation.

**Comment by Barryrowe** [comment]
Yeah, this is going to be a tough to to track down since they all run fine through wallaby. Doesn't look like it's a jasmine version issue.  Thanks @ArtemGovorov

**Comment by Barryrowe** [comment]
For anyone that might run into something similar I sorted this out. It has nothing to do with Wallaby really, other than that Wallaby happened to isolate tests in a way that the issue never presented itself. 

TL;DR; Some of my Async tests had issues, which Karma apparently allowed to affect tests down the line, where Wallaby seems to have isolated them so there were no issues (just a guess).

It was a combination of a few things. This occurred during an upgrade from Ionici2@beta.11 to Ionic2@rc.3, which is also an upgrade from Angular2@rc.4 to Angular@^2.1.1

The root of the issue is that my async tests were not using an async mechanism, and what had worked prior in the Angular wrapped Jasmin functions (it, expect, etc.) no longer was reliable.  Before I was able to return a promise from an ```it``` and the test would recognize it as async and I guess handle the done() call internally.

I had to switch all of my tests from this format:

``` javascript
describe("Simple Serivce", () => {
    beforeEach(() => TestBed.configureTestingModule({
         providers: [ MyService, {provide: OtherService, useClass: MockOtherService}]
    }));

    it("should do an async thing", inject[MyService, OtherService], (ms:MyService, os:OtherService) => {
        spyOn(os, "expectedMethodCall");
        return ms.doAsyncThing()
                       .then(() => expect(os.expectedMethodCall).toHaveBeenCalled() );
    }));
});
```
to

``` javascript
describe("Simple Serivce", () => {
    let ms: MyService,
         os: OtherService;

    beforeEach(() => {
        TestBed.configureTestingModule({
             providers: [ MyService, {provide: OtherService, useClass: MockOtherService}]
        });
         inject[MyService, OtherService], (myService:MyService, otherService:OtherService) => {
               ms = myService;
               os = otherService;
         })();
    
});
it("should do an async thing", (done) => {
    spyOn(os, "expectedMethodCall");
    return ms.doAsyncThing()
                   .then(() => {
                        expect(os.expectedMethodCall).toHaveBeenCalled();
                        done();
                   });
});
```

In the end, I'm much happier with the structure of these tests, as they look like standard Jasmine tests, and less like some specific way of testing Angular stuff with the inject() in the it blocks.

**Comment by ArtemGovorov** [comment]
Wow, thanks for sharing!
Wallaby runs tests in a few parallel chunks (each one is isolated). You can configure it to work more as karma (in terms of isolation) by disabling the parallelisation [by setting the `workers` count to `1`](https://wallabyjs.com/docs/config/workers.html).
